homeworks/olena.ambitska_elena-ambitska/GamesProject/scripts/login.js

```diff
@@ -0,0 +1,107 @@
+const VALIDATE_RULES = {
+    email: {
+        rule: (email) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email),
+        message: 'Invalid email format',
+        required: true,
+    },
+    password: {
+        rule: (password) => password.length >= 8 && password.length <= 12,
+        message: 'Password must be between 8 and 12 characters',
+        required: true,
+    },
+    checkbox: {
+        rule: (checkbox) => checkbox,
+        message: 'Checkbox must be checked',
+        required: true,
+    },
+};
+
+const ERROR_STATE = Object.fromEntries(Object.keys(VALIDATE_RULES).map((key) => [key, false]));
+
+function displayErrorMessage(field, message) {
+    const errorContainer = document.createElement('div');
+    errorContainer.classList.add('error-message');
+    errorContainer.setAttribute('data-field', field);
+    errorContainer.innerText = message;
+
+    const inputField = document.getElementById(field);
+    const formGroup = inputField.closest('.form-group');
+    formGroup.appendChild(errorContainer);
+}
+
+function validateField(field) {
+    const value = document.getElementById(field).value.trim();
+    const rule = VALIDATE_RULES[field];
+    const isChecked = document.getElementById(field).checked;
+
+    const isFieldEmpty = rule.required && !value;
+    const isCheckboxInvalid = field === 'checkbox' && !rule.rule(isChecked);
+    const isValueInvalid = rule.rule && !rule.rule(value);
+
+    // eslint-disable-next-line object-curly-newline
```

Comment:
you have two functions with similar names: `validateField` and `validateInputField`
If the current function returns a validity state for the given input, I propose to rename it to `getFieldValidity`. It would help to prevent messing up with the names.

Reply:
@vladyslav-yermolin-moc done for every thread. 
1. Just a habit :see_no_evil: